test/libsolidity/syntaxTests/enums/enum_member_shadowing.sol

```diff
@@ -0,0 +1,12 @@
+contract C {
+    struct S {
+        uint x;
+    }
+
+    enum E {E, S, C, a, f}
+
+    uint a;
+
+    function f() public pure {}
+}
+// ----
```

Comment:
Shouldn't this issue a compiler error? Because enum member shadows type name?

Reply:
No, because shadowing is legal. It's only when you shadow type names used later within the same scope that it's an error:
```solidity
{
    Type1 Type2 = 1;
    Type2 x = 2;
}
```
You cannot do that in an enum. I added this test mostly for completeness.